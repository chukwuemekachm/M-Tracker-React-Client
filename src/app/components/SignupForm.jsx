import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      email: '',
      password: '',
      confirmPassword: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  onSubmit(event) {
    event.preventDefault();
    const { password, confirmPassword } = this.state;
    const { userSignupRequest, userError, history } = this.props;
    if (confirmPassword !== password) {
      userError('Passwords do not match');
      return;
    }
    userSignupRequest(this.state, history);
  }

  render() {
    const {
      firstname, lastname, email, password, confirmPassword,
    } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <label htmlFor="firstName" className="ch-label">
          First Name
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="text"
            name="firstname"
            className="ch-input"
            required
            placeholder="First Name"
            value={firstname}
            onChange={this.onChange}
          />
        </label>
        <label htmlFor="lastname" className="ch-label">
          Last Name
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="text"
            name="lastname"
            className="ch-input"
            required
            placeholder="Last Name"
            value={lastname}
            onChange={this.onChange}
          />
        </label>
        <label htmlFor="email" className="ch-label">
          Email Address
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="email"
            name="email"
            className="ch-input"
            required
            placeholder="Email Address"
            value={email}
            onChange={this.onChange}
          />
        </label>
        <label htmlFor="password" className="ch-label">
          Password
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="password"
            name="password"
            className="ch-input"
            required
            placeholder="Password"
            value={password}
            onChange={this.onChange}
          />
        </label>
        <label htmlFor="confirmPassword" className="ch-label">
          Confirm Password
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="password"
            name="confirmPassword"
            className="ch-input"
            required
            placeholder="Password"
            value={confirmPassword}
            onChange={this.onChange}
          />
        </label>
        <button type="submit" className="ch-btn-flat">
          Signup
        </button>
      </form>
    );
  }
}

SignupForm.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
  userError: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};

export default SignupForm;
