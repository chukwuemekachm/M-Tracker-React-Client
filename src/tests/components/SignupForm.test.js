import React from 'react';
import { mount } from 'enzyme';
import SignupForm from '../../app/components/SignupForm';

const userError = jest.fn();
const userSignupRequest = jest.fn();
const history = {};
describe('Navbar component', () => {
  it('should render without errors', () => {
    const wrapper = mount(<SignupForm
      userError={userError}
      userSignupRequest={userSignupRequest}
      history={history}
    />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('[name="firstname"]').simulate('change', {
      target: { value: 'john' },
    });
    wrapper.find('[name="lastname"]').simulate('change', {
      target: { value: 'doe' },
    });
    wrapper.find('[name="email"]').simulate('change', {
      target: { value: 'john@gmail.com' },
    });
    wrapper.find('[name="password"]').simulate('change', {
      target: { value: 'Password' },
    });
    wrapper.find('[name="confirmPassword"]').simulate('change', {
      target: { value: 'Password' },
    });
    wrapper.find('button').simulate('click');
  });
});
