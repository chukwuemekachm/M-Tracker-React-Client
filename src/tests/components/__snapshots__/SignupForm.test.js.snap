// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render without errors 1`] = `
<SignupForm
  history={Object {}}
  userError={[MockFunction]}
  userSignupRequest={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="ch-label"
      htmlFor="firstName"
    >
      First Name
       
      <span
        className="ch-require"
      >
        *
      </span>
      <input
        className="ch-input"
        name="firstname"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      className="ch-label"
      htmlFor="lastname"
    >
      Last Name
       
      <span
        className="ch-require"
      >
        *
      </span>
      <input
        className="ch-input"
        name="lastname"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      className="ch-label"
      htmlFor="email"
    >
      Email Address
       
      <span
        className="ch-require"
      >
        *
      </span>
      <input
        className="ch-input"
        name="email"
        onChange={[Function]}
        placeholder="Email Address"
        required={true}
        type="email"
        value=""
      />
    </label>
    <label
      className="ch-label"
      htmlFor="password"
    >
      Password
       
      <span
        className="ch-require"
      >
        *
      </span>
      <input
        className="ch-input"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </label>
    <label
      className="ch-label"
      htmlFor="confirmPassword"
    >
      Confirm Password
       
      <span
        className="ch-require"
      >
        *
      </span>
      <input
        className="ch-input"
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </label>
    <button
      className="ch-btn-flat"
      type="submit"
    >
      Signup
    </button>
  </form>
</SignupForm>
`;
